File: RealEstate.Domain/Entities/Property.cs
using RealEstate.Domain.Enums;

namespace RealEstate.Domain.Entities
{
    public class Property
    {
        public Guid Id { get; set; }
        public string Title { get; set; } = default!;
        public string Address { get; set; } = default!;
        public string Location { get; set; } = default!;
        public string City { get; set; } = default!;
        public decimal Price { get; set; }
        public ListingType ListingType { get; set; }
        public int Bedrooms { get; set; }
        public int Bathrooms { get; set; }
        public int CarSpots { get; set; }
        public string SizeLabel { get; set; }
        public string AreaLabel { get; set; }
        public string Status { get; set; }
        public string ThumbnailUrl { get; set; }
        public string Description { get; set; } = default!;
        public List<string> ImageUrls { get; set; } = new();
        public double? Lat { get; set; }
        public double? Lng { get; set; }
        public string PropertyType { get; set; }
        public ICollection<Favorite> FavoritedBy { get; set; } = new List<Favorite>();
    }
}

File: frontend/src/types/property.ts
import type { FavoriteDto } from "./favorite";
import type { ListingType } from "./listing-type";

export type PropertyDto = {
  id: string;
  title: string;
  address?: string;
  location?: string;
  city?: string;
  price: number;
  bedrooms?: number | null;
  bathrooms?: number | null;
  carSpots?: number | null;
  sizeLabel?: string; 
  areaLabel?: string;
  status?: "For Sale" | "For Rent" | string;
  thumbnailUrl?: string;
  imageUrls?: string[];
  listingType: ListingType;
  favoritedBy: FavoriteDto[];
  isFavorite?: boolean;
  lat?: number | null;
  lng?: number | null;
  propertyType?: "All" | "House" | "Apartment" | "Plot" | string;
};

export type PropertyQuery = {
  q?: string;
  city?: string;
  type?: "All" | "House" | "Apartment" | "Plot" | "";
  minPrice?: number | null;
  maxPrice?: number | null;
  bedrooms?: number | null;
  bathrooms?: number | null;
  sortBy: string;
  sortOrder: "asc" | "desc";
  page: number;
  pageSize: number;
};
